packages/web/server/queries/complex/assets/index.ts

```diff
@@ -66,231 +58,73 @@ export async function getAssets({
   /** Explicitly match the base or symbol denom. */
   findMinDenomOrSymbol?: string;
 } & AssetFilter = {}): Promise<Asset[]> {
-  const makeMinimalAssets = (assetList: AssetList[]) => {
-    // Create new array with just assets
-    const coinMinimalDenomSet = new Set<string>();
-
-    const listedAssets = assetList
-      .flatMap(({ assets }) => assets)
-      .filter(
-        (asset) =>
-          asset.keywords && !asset.keywords.includes("osmosis-unlisted")
-      );
-
-    let assetListAssets = listedAssets.filter((asset) => {
-      if (params.findMinDenomOrSymbol) {
-        return (
-          params.findMinDenomOrSymbol.toUpperCase() ===
-            asset.base.toUpperCase() ||
-          params.findMinDenomOrSymbol.toUpperCase() ===
-            asset.symbol.toUpperCase()
-        );
-      }
-
-      // Ensure denoms are unique on Osmosis chain
-      // In the case the asset list has the same asset twice
-      if (coinMinimalDenomSet.has(asset.base)) {
-        return false;
-      } else {
-        coinMinimalDenomSet.add(asset.base);
-        return true;
-      }
-    });
-
-    // Search raw asset list before reducing type to minimal Asset type
-    if (params.search) {
-      assetListAssets = search(
-        assetListAssets,
-        searchableAssetListAssetKeys,
-        params.search
-      );
-    }
-
-    // Filter by only verified
-    if (params.onlyVerified) {
-      assetListAssets = assetListAssets.filter((asset) =>
-        Boolean(asset.keywords?.includes("osmosis-main"))
-      );
-    }
-
-    // Transform into a more compact object
-    return assetListAssets.map(makeMinimalAsset);
-  };
-
   // if it's the default asset list, cache it
   if (assetList === AssetLists) {
     return cachified({
       cache: minimalAssetsCache,
       key: JSON.stringify(params),
-      getFreshValue: () => makeMinimalAssets(assetList),
+      getFreshValue: () => reduceAssetList(assetList, params),
     });
   }
 
   // otherwise process the given novel asset list
-  return makeMinimalAssets(assetList);
+  return reduceAssetList(assetList, params);
 }
 
-/** Basic user info if the user holds a balance. */
-export type MaybeUserAssetInfo = Partial<{
-  amount: CoinPretty;
-  usdValue: PricePretty;
-}>;
-
-export async function getUserAssetInfo<TAsset extends Asset>({
-  assetList = AssetLists,
-  asset,
-  userOsmoAddress,
-}: {
-  assetList?: AssetList[];
-  asset: TAsset;
-  userOsmoAddress?: string;
-}): Promise<TAsset & MaybeUserAssetInfo> {
-  if (!userOsmoAddress) return asset;
+/** Transform given asset list into an array of minimal asset types for user in frontend. */
+function reduceAssetList(
```

Comment:
What do you think about renaming this function to `simplifyAssetListForDisplay`?